hera_cal/vis_clean.py

```diff
@@ -281,22 +307,419 @@ def write_data(self, data, filename, overwrite=False, flags=None, nsamples=None,
+                    ax: str, axis to filter, options=['freq', 'time', 'both']
+                *'clean':
+                     'tol': float,
+                        clean tolerance. 1e-9 is standard.
+                     'maxiter' : int
+                        maximum number of clean iterations. 100 is standard.
+                     'pad': int or array-like
+                        if filt2d is false, just an integer specifing the number of channels
+                        to pad for CLEAN (sets Fourier interpolation resolution).
+                        if filt2d is true, specify 2-tuple in both dimensions.
+                     'filt2d_mode' : string
+                        if 'rect', clean withing a rectangular region of Fourier space given
+                        by the intersection of each set of windows.
+                        if 'plus' only clean the plus-shaped shape along
+                        zero-delay and fringe rate.
+                    'edgecut_low' : int, number of bins to consider zero-padded at low-side of the FFT axis,
+                        such that the windowing function smoothly approaches zero. For 2D cleaning, can
+                        be fed as a tuple specifying edgecut_low for first and second FFT axis.
+                    'edgecut_hi' : int, number of bins to consider zero-padded at high-side of the FFT axis,
+                        such that the windowing function smoothly approaches zero. For 2D cleaning, can
+                        be fed as a tuple specifying edgecut_hi for first and second FFT axis.
+                    'add_clean_residual' : bool, if True, adds the CLEAN residual within the CLEAN bounds
+                        in fourier space to the CLEAN model. Note that the residual actually returned is
+                        not the CLEAN residual, but the residual in input data space.
+                    'taper' : window function for filtering applied to the filtered axis.
+                        See dspec.gen_window for options. If clean2D, can be fed as a list
+                        specifying the window for each axis in data.
+                    'skip_wgt' : skips filtering rows with very low total weight (unflagged fraction ~< skip_wgt).
+                        Model is left as 0s, residual is left as data, and info is {'skipped': True} for that
+                        time. Only works properly when all weights are all between 0 and 1.
+                    'gain': The fraction of a residual used in each iteration. If this is too low, clean takes
+                        unnecessarily long. If it is too high, clean does a poor job of deconvolving.
+                    'alpha': float, if window is 'tukey', this is its alpha parameter.
+        x : array-like, x-values of axes to be filtered. Numpy array if 1d filter.
+            2-list/tuple of numpy arrays if 2d filter.
+     """
+        if not HAVE_UVTOOLS:
```

Comment:
uvtools is a hera_cal prereq. This is unnecessary.

Reply:
I'll remove this (and other instances from previous commits).